velox/functions/prestosql/aggregates/MaxSizeForStatsAggregate.cpp

```diff
@@ -0,0 +1,243 @@
+  return serde;
+}
+} // namespace
+
+namespace {
+
+class MaxSizeForStatsAggregate
+    : public SimpleNumericAggregate<int64_t, int64_t, int64_t> {
+  using BaseAggregate = SimpleNumericAggregate<int64_t, int64_t, int64_t>;
+
+ public:
+  explicit MaxSizeForStatsAggregate(TypePtr resultType)
+      : BaseAggregate(resultType) {}
+
+  int32_t accumulatorFixedWidthSize() const override {
+    return sizeof(int64_t);
+  }
+
+  void extractValues(char** groups, int32_t numGroups, VectorPtr* result)
+      override {
+    BaseAggregate::doExtractValues(groups, numGroups, result, [&](char* group) {
+      return *BaseAggregate::Aggregate::template value<int64_t>(group);
+    });
+  }
+
+  void initializeNewGroups(
+      char** groups,
+      folly::Range<const vector_size_t*> indices) override {
+    exec::Aggregate::setAllNulls(groups, indices);
+    for (auto i : indices) {
+      *BaseAggregate ::value<int64_t>(groups[i]) = 0;
+    }
+  }
+
+  void extractAccumulators(char** groups, int32_t numGroups, VectorPtr* result)
+      override {
+    // Partial and final aggregations are the same.
+    extractValues(groups, numGroups, result);
+  }
+
+  void addIntermediateResults(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool mayPushdown) override {
+    BaseAggregate::template updateGroups<true, int64_t>(
+        groups,
+        rows,
+        args[0],
+        [](int64_t& result, int64_t value) {
+          if (result < value) {
+            result = value;
+          }
+        },
+        mayPushdown);
+  }
+
+  void addSingleGroupIntermediateResults(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool mayPushdown) override {
+    BaseAggregate::updateOneGroup(
+        group,
+        rows,
+        args[0],
+        [](int64_t& result, int64_t value) {
+          result = result > value ? result : value;
+        },
+        [](int64_t& result, int64_t value, int /* unused */) {
+          result = value;
+        },
+        mayPushdown,
+        (int64_t)0);
+  }
+
+  void addSingleGroupRawInput(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    doUpdateSingleGroup(group, rows, args[0]);
+  }
+
+  void addRawInput(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    doUpdate(groups, rows, args[0]);
+  }
+
+ protected:
+  void upsertOneAccumulator(
+      char* const group,
+      const BaseVector& inputVector,
+      std::vector<vector_size_t>& rowSizes,
+      vector_size_t idx) {
+    if (inputVector.isNullAt(idx)) {
+      return;
+    }
+
+    // Clear null.
+    clearNull(group);
+    // Set max(current, this).
+    int64_t& current = *value<int64_t>(group);
+    current = std::max(current, (int64_t)rowSizes[idx]);
```

Comment:
consistency: The code appear inconsistent as it uses std::max here, but not in addSingleGroupIntermediateResults. It would be nice to use the same code in both places.